gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -238,10 +253,46 @@ private Completable handlePhotoCaptureResult(String fieldId) {
         .ignoreElements();
   }
 
-  private Observable<File> saveBitmapAndUpdateResponse(Observable<Bitmap> source, String fieldId) {
+  private Observable<String> saveBitmapAndUpdateResponse(
+      Observable<Bitmap> source, String fieldId) {
     return source
         .map(bitmap -> fileUtil.saveBitmap(bitmap, fieldId + ".jpg"))
-        .doOnNext(file -> onTextChanged(form.getValue().getField(fieldId).get(), file.getPath()));
+        .map(
```

Comment:
Are we filtering by JPG?

Also, is it possible to upload more than one image for a particular field? If so `fieldId` alone won't unique.

Reply:
No, we save the bitmap as a .jpg file and that gets uploaded. 

Currently, one field can have only one photo. So, the old one gets overwritten if re-uploaded.

Reply:
I think that's ok as a starting point. We can see if more are needed. Thanks!

Reply:
I also wonder if we'll get into trouble if users uploading PNG and GIF from their device - let's remember test this.